ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/config/DataEdgeConfig.java

```diff
@@ -35,4 +69,134 @@ public TaskExecutor getAsyncExecutor() {
+        return new JWTServiceImpl(new JWKSet(), "rsa1");
+    }
+
+    @Bean
+    public FileDatasetService initFileDatasetServiceImpl(final RestTemplate restTemplate) {
+        return new FileDatasetServiceImpl(restTemplate);
+    }
+
+    @Bean
+    public UserDetailsService initUserDetailsServiceImpl(final RestTemplate restTemplate) {
+        return new UserDetailsServiceImpl(restTemplate);
+    }
+
+    /**
+     * This is a bit of a Hack! MitreID doesn't return 'user_name' but 'user_id',
+     * The customized User Authentication Converter simply changes the field name
+     * for extraction.
+     *
+     * @return The converted access token
+     */
+    @Bean
+    public AccessTokenConverter accessTokenConverter(final JWTService jwtService,
+                                                     final Ga4ghService ga4ghService,
+                                                     final FileDatasetService fileDatasetService,
+                                                     final UserDetailsService userDetailsService) {
+        return new MyAccessTokenConverter(
+                jwtService,
+                ga4ghService,
+                fileDatasetService,
+                new MyUserAuthenticationConverter(),
+                userDetailsService
+        );
+    }
+
+    /**
+     * Creates a combined token service including both EGA AAI and Elixir AAI
+     * authentication.
+     *
+     * @param request Unused
+     * @param checkTokenUrl EGA AAI token endpoint url
+     * @param clientId Client id for the EGA AAI
+     * @param clientSecret Client secret for the EGA AAI
+     * @param zuulCheckTokenUrl Elixir token endpoint url
+     * @param zuulClientId Elixir AAI ID
+     * @param zuulClientSecret Elixir AAI client
+     *
+     * @return A combined authentication token service
+     */
+    @Profile("enable-aai")
+    @Primary
+    @Bean
+    public RemoteTokenServices remoteTokenServices(HttpServletRequest request,
+                                                   final @Value("${auth.server.url}") String checkTokenUrl,
+                                                   final @Value("${auth.server.clientId}") String clientId,
+                                                   final @Value("${auth.server.clientsecret}") String clientSecret,
+                                                   final @Value("${auth.zuul.server.url}") String zuulCheckTokenUrl,
+                                                   final @Value("${auth.zuul.server.clientId}") String zuulClientId,
+                                                   final @Value("${auth.zuul.server.clientsecret}") String zuulClientSecret,
+                                                   final AccessTokenConverter accessTokenConverter) {
+
+        final CachingMultipleRemoteTokenService remoteTokenServices = new CachingMultipleRemoteTokenService();
+
+        // EGA AAI
+        CachingRemoteTokenService b = new CachingRemoteTokenService();
```

Comment:
Could we give a meaningful name of the variables? It would help to understand better the code :)

Reply:
Done.